### PR TITLE
[SPARK-7227][SPARKR] Support fillna / dropna in R DataFrame.

### PR DESCRIPTION
**Pull request by sun-rui**

### CODE REVIEW

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1431,3 +1431,119 @@ setMethod("describe",
...
+#' @return A DataFrame
+#' 
+#' @rdname nafunctions
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: Could you also add an example here how how the named list usage looks ?

File: core/src/main/scala/org/apache/spark/api/r/SerDe.scala [hunk 2]
```diff
@@ -166,6 +166,24 @@ private[spark] object SerDe {
     }
   }
 
+  def readEnv(in: DataInputStream): java.util.Map[Object, Object] = {
```
Comment (shivaram) [inline core/src/main/scala/org/apache/spark/api/r/SerDe.scala]: Minor comment - R env's are restricted to have String keys, so readMap is still a good name as this is probably a bit more general than R environments ?

**Comment by shivaram** [comment]
Thanks @sun-rui. This is looking very good ! I had some very minor comments

@rxin could you also take a look at the semantics and make sure it looks ok ?

**Comment by rxin** [comment]
seems fine to me.

**Comment by shivaram** [comment]
@sun-rui one more comment -- lets alias `na.omit` to `dropna` -- That is the method used for local R data frames https://stat.ethz.ch/R-manual/R-patched/library/stats/html/na.fail.html

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: OK I will do it

File: core/src/main/scala/org/apache/spark/api/r/SerDe.scala (hunk 2 above)
Comment (sun-rui) [inline core/src/main/scala/org/apache/spark/api/r/SerDe.scala]: I just want to keep readMap() for the case where in R side an environment in which all values are of same type, the value type is serialized once. It is optimized than readEnv(). But I am OK if you think we can update readMap() and remove readEnv() for now and do such optimization in the future if necessary.

Comment (shivaram) [inline core/src/main/scala/org/apache/spark/api/r/SerDe.scala]: My comment was only about the name for the function. It doesn't matter actually as its just an internal function.

**Comment by rxin** [comment]
@shivaram what's happening with this one?

**Comment by shivaram** [comment]
I thought I was waiting for a update from @sun-rui, but actually my comments have been addressed. There is a minor R documentation change, but I can do that with the merge. 

LGTM.

**Pull request closed without merging**